### DISCUSSION TITLE
Slack or similar?

### DISCUSSION THREAD

**Discussion by vsoch** [hpc-social/hpc-social.github.io#52] [Q&A] [answered]
hey @FCLC and @alansill ! I was asked today if there  is a plan for some kind of chat space for hpc.social - e.g., a channel in an existing slack or a new slack. Any thoughts? I think right now I'm using Twitter (group chat) and Mastodon DM for small discussion, but the mastodon DM isn't great.

**Comment by alansill** [comment]
I've never found out how to support using slack for people from multiple organizations without facing very high costs. Is there a way?

Moderating a slack channel sounds hard if it's open to everyone.

**Reply by vsoch** [reply]
Maybe piggy back off of a paid slack? or just be like spack slack and stay on the free tier and have a lot of messages go away (I think it's like 60 or 90 days). I guess as long as the space is thought of as ephemeral, the free tier sort of works.

Pinging @davidbeckingsale since we were chatting about this!

**Reply by FCLC** [reply]
(Group) DM on mastodon, especially via the web UI is pretty bad. 

DM via some of the apps is a decent experience. MetaText on iOS gets it mostly right. Haven't tested group DM via meta text, you'll probably see something from me in a second as a test. 

If the idea is group messaging organization at low/no cost the easiest solutions that come to mind are:
A Discord server
Setting/hosting our own Matrix instance*

Of the two I think Discord is probably the easiest. I know BLIS uses it amongst other projects (for the same cost issues above)

We run into similar issues with twitter where we aren't in control of our own destiny per say, and whatever we choose now will be likely to stick longer term. I did a quick search around and I don't see any active development on an activity pub compatible, discord/slack/teams like interface/project/"thing". 

If/when that comes along it may be a great solution. 



*(unfortunately a different protocol to activity pub, which is what powers mastodon, and potentially another point of friction for the community).

**Reply by vsoch** [reply]
I'm actually on element (which is matrix) because that's where the nix community is - I think it's less popular than say, slack, but I definitely don't mind it. I have issues with discord and what they were doing with personal info a while back and I don't use it, period, so I'd rather not add it back. Context: https://www.reddit.com/r/privacy/comments/rsxeee/you_should_never_use_discord_and_heres_why/

**Reply by FCLC** [reply]
![image](https://user-images.githubusercontent.com/20584423/207730861-a82be690-398a-426f-8ef2-34c1182f0f04.jpeg)

**Reply by FCLC** [reply]
Yeah, I get what you mean with Discord; it's far from great. 

I feel like Slack as a middle ground is fine, but loosing data feels very un-HPC. 

~~I'd like to avoid teams like the plague it is.~~

@alansill What's your thinking on Matrix/element? I've used it for some of the online conferences (XDC 22 used it for example) and it's mostly good, but if we were already facing friction with mastodon, I feel like asking users to join Yet Another App isn't going to work out very well.

**Reply by vsoch** [reply]
What's wrong with slack? A lot of big HPC projects use it, at least, and the free tier is really fine for what we generally need. Just don't expect conversation to stay there after 60 or 90 days.

**Reply by vsoch** [reply]
We could put a poll out to the community to ask maybe? Even discussions I think now supports them!

**Reply by alansill** [reply]
Just the Spack slack has like 2k users. I find it very hard to find or keep track of things there even when I'm interested in them. how would we organize/monitor/moderate/answer questions on a Slack instance?

**Reply by alansill** [reply]
I guess I'm wonderiing what problem for the community we're solving by bringing up a chat instance beyond the Mastodon server itself.

**Reply by alansill** [reply]
Here's perhaps a prototypical example of a forum that might work. It appears to be using MatterMost, so we'd have to find a place to host it and a way to cover the costs. At least it appears to have the right features and looks way more organized than the mushpool of stream of consciousness threads that I see when I look at Slack channels.  https://forums.rockylinux.org

**Reply by alansill** [reply]
They have a non-profit/open source/educational plan https://docs.mattermost.com/about/subscription.html#do-you-have-a-program-for-official-non-profits-open-source-projects-and-charities

**Reply by vsoch** [reply]
oh yeah! So I use mattermost for my employer - it's definitely good, sort of like slack but just a slightly different feel.

**Reply by vsoch** [reply]
> I guess I'm wonderiing what problem for the community we're solving by bringing up a chat instance beyond the Mastodon server itself.

I like chatting with my colleagues, or people in my community. We currently don't have a place to do that. That's the need I think.

**Reply by vsoch** [reply]
@alansill a forum is different than a chat server - I don't think we need a forum because we have GitHub discussions, but chat is different. The one you linked for Rocky is discourse I think, and it's what https://ask.cyberinfrastructure.org/ uses too. I've never really loved it.

**Comment by alansill** [comment]
By coincidence this was posted on Tecmint today: https://www.tecmint.com/slack-alternatives/

**Reply by vsoch** [reply]
Oh wow, the number of options is overwhelming. :sob:

**Comment by sramkrishna** [comment]
So since we're already doing the de-centralized thing - I would suggest we use matrix - (https://matrix.org/) - lots of choices on clients. The server is open source and the matrix guys are cool. I'm in many communities of course not just HPC and the GNOME and KDE open source desktop communities have switched to using matrix many in lieu of IRC. You can recreate the irc feeling and also do the slack thing.

Secondly, you can attach services like BigBlueButton - at FOSDEM I believe they used this combination to run FOSDEM remotely (for those who know) I highly suggest matrix, although there are several choices for video conferencing - but I enjoy using BigBlueButton.

**Reply by sramkrishna** [reply]
oh yeah, you don't have to use your own server - you could use the one that is supported by matrix.org - especially if you don't want to deal with administration. 

If you have an existing matrix account then you can just use that but otherwise you can have a sri@hpc.social - identity. Channels could be something like #blas:hpc.social.

**Reply by vsoch** [reply]
I would +1 this! I mentioned this above, but I use matrix via element, and the Nix communities are on there.

**Reply by sramkrishna** [reply]
Right and you can easily connect to those communities from element - you could I think even bridge to other communities. It's possible to link to telegram as well but I don't trust that much. Of course encryption all the way.

**Comment by alansill** [comment]
For now, could we point to ask.ci and cnct.ci instead of making our own forums? But forums are not chat - or are they? I'd hate to duplicate effort but am happy to entertain ideas to fill a need.

**Reply by vsoch** [reply]
No forums are not chat - we already have these discussion areas for this kind of “forum discussion” but we don’t have a place to chat :)

**Comment by alansill** [comment]
Hmm, matrix.org does look nice as a platform. Also many [bridges exist](https://matrix.org/bridges/).  Hmm.

We'd need sponsors to do anything serious I think.

**Reply by vsoch** [reply]
Why not use element for matrix? https://app.element.io Say the word and  I'll try making us an hpc.social channel!

**Reply by alansill** [reply]
Sounds okay as an experiment.

**Reply by alansill** [reply]
Is there a way to use Mastodon identities from e.g. mast.hpc.social to get in so people don't have to make yet another account? What will we do about moderation etc.? What administrative activities are appropriate / required / essential or desirable to make the association with hpc.social official and in keeping with our usage guidelines and mission?

**Reply by vsoch** [reply]
I don't know. But I suspect others have thought about it. https://www.reddit.com/r/Mastodon/comments/mu7z5d/provide_matrix_service_to_mastodon_users/

**Reply by sramkrishna** [reply]
Yes, it's worth considering. I think matrix does support single sign on which I think is what we are looking at. But the complication in this case is that people who have existing identities on matrix would likely prefer to use those. So it might get messy.

**Comment by alansill** [comment]
Based on the discussion above the cleanest thing to do seems to be to start an hpc.social Slack workspace and use it to experiment with integrations or to see if it can grow to useful levels of communication on its own. One has now been created. Let me know here if you would like an invitation to join.

**Reply by vsoch** [reply]
Can you send me a link somewhere to join?

**Reply by alansill** [reply]
There's one in our Twitter group chat for now, can send to your email address that I think I have somewhere if you prefer.

**Reply by sramkrishna** [reply]
I can understand if that is the cleanest and simplest I just hate giving slack any kind of credence because they aren't open source and their claim to fame is that they manage to make a better irc. I prefer trusting myself to pure community related projects. But if you're trying to capture momentum. I might make it a personal project of mine to see if I can make it happen on matrix - perhaps myself and @vsoch could collaborate since we are the main proponents right now. I just see a brighter future than slack :D

**Reply by hestes835** [reply]
> I can understand if that is the cleanest and simplest I just hate giving slack any kind of credence because they aren't open source and their claim to fame is that they manage to make a better irc. I prefer trusting myself to pure community related projects. But if you're trying to capture momentum. I might make it a personal project of mine to see if I can make it happen on matrix - perhaps myself and @vsoch could collaborate since we are the main proponents right now. I just see a brighter future than slack :D

Did anything come from this? I would love to join a matrix server with everyone

**Comment by sramkrishna** [comment]
I need a place to host and we will need to figure out costs. For now I
could do a aws thing

**Reply by alansill** [reply]
I've investigated this and am trying to decide among approaches.

**Comment by sramkrishna** [comment]
Alan,
You could just go with the one that is hosted by matrix.org as a channel as
a start. I don't think there is a cost there but a good way to kick the
tires? In any case, I'm sure you know how to go about this.

**Reply by alansill** [reply]
More information and links from the matrix.org organization: https://matrix.org/blog/2022/12/25/the-matrix-holiday-update-2022/

**Reply by alansill** [reply]
The Slack instance is getting uptake based on having people request invitations, and the Discord instance option to include the SC22 rooms suggested by @Qeolvdxu Seems to be popular there. What is the best way for us to roll these out as separate options while we look into the options (which seem to be expensive so far) for bridging and/or integration? I assume these should emerge onto the hpc.social site as projects under the "Chat" category or something similar?

**Reply by vsoch** [reply]
I would probably just link invite links (or metadata) in the .github README, I'm not sure they are established projects yet.

**Reply by alansill** [reply]
Okay, I asked Hayden to contact you. I am okay with one project card for hpc.social chat services like the other cards that we have that gives a short explanation and lists links to get an invitation to each. Since you started the Slack creation, I think you should be the point person to set this up but let me know if I can help with this or announcements, etc.

He likes the dino icon and if so we should probably rebrand the Slack instance with that also. (But I liked the inverse hpc.social one so if thereʻs any way to include that in the branding somehow Iʻd like that too.)

**Reply by vsoch** [reply]
Sure I can create that page. I’m finishing up some Kubernetes work now and actively running services (that I’m paying for) so I want to do that first, but I’ll have time for this worst case over the weekend.

**Reply by vsoch** [reply]
okay - I updated the slack icon! I can start the "chat spaces" page, but I'll wait for Hayden to reach out so I know more about Discord. We will also need a way for users to get an invite (an enduring invite link requires a tiny server running somewhere, unless you know another way, and also I think we should somewhat curate who is asking) so in the meantime what do you think - ask them to ping one of us on Twitter? In GitHub discussions /issues? mastodon or something else? (all of the above?)

**Reply by vsoch** [reply]
And here is the start of the chats page - we can add Discord if/when that information is available. https://github.com/hpc-social/hpc-social.github.io/pull/55

**Comment by vsoch** [comment, marked as answer]
We now have a series of chat clients for the HPC social community, which you can find an updated list on https://hpc.social/projects/chat/. We also have bots that can post there, currently for bots. Please add to this thread if you'd like to discuss a new option!

**Comment by sramkrishna** [comment]
Given where we are with slack, do we want to revisit and continue to use slack or should we move elsewhere (plus I believe they believe they can use our content for AI training)

**Reply by vsoch** [reply]
My preference is to use slack - it continues to get new people added, is already substantially sized, and seems to be working well. Personally speaking, I don't really have interest to add another place to chat (I would not join).

**Reply by alansill** [reply]
I agree. The usage is pretty casual so unless the AI is mining for some excellent shared humor, I think the community is pretty happy where we are. Other opinions of course are welcome and if there is concern, we can respond accordingly.

**Reply by FCLC** [reply]
We can also send an opt out request to slack? 

It has to be the workspace Owner (I think that's @alansill ?)

A single email and we're opted out 

Edit: link to instructions/relevant information: https://slack.com/trust/data-management/privacy-principles

**Reply by alansill** [reply]
I think @vsoch is the workspace owner for Slack.

**Reply by vsoch** [reply]
Done.

**Comment by sramkrishna** [comment]
OK - fair, I was just hearing that there were going to start charging per user or something and I got worried.
